### ISSUE TITLE
DOC Document how to use micropip.freeze

### ISSUE THREAD

**Issue by rth** [pyodide/pyodide#2948]
We should add documentation about how to use  [micropip.freeze](https://pyodide.org/en/latest/usage/api/micropip-api.html?micropip.freeze) to https://pyodide.org/en/latest/usage/loading-packages.html#micropip.

Also its docstring states,
>  Loading packages with [pyodide.loadPackage](https://pyodide.org/en/latest/usage/api/js-api.html#pyodide.loadPackage) is much faster and you will always get consistent versions of all your dependencies.

I have several comments about this,
 a) is this really much faster?  -- I don't think so. We are still loading the same wheel just without the need to load micropip first.
 b) Anyway we should maybe emphasize less how loadPackage is better than micropip here, because we have been telling being people the opposite so far. Maybe word it rather in terms of loadPackage being a lower level command which limited functionality but a bit less overhead.
 c) also "consistent"  -> fixed.

**Comment by hoodmane** [comment]
> is this really much faster? -- I don't think so

Well in the best case `micropip` takes two sequential requests to download a wheel: 
1. https://pypi.org/pypi/pkg-name/json
2. https://files.pythonhosted.org/packages/ha/sh/more-hash/pkg-name.whl

Once the wheel is fully downloaded, we discover what dependencies the wheel has. Each dependency also requires two requests (if it isn't in the lock file). If pkg A depends on pkg B depends on pkg C, we need 6 sequential requests.
If we need to download micropip for this then that's one extra request, so downloading a single wheel with no dependencies then takes 3 sequential requests.

With the frozen version we can make all requests concurrently so it only takes 1 request round trip.

**Comment by hoodmane** [comment]
Also, it should be possible in the future to download some dependencies from a lock file while Pyodide is loading, in which case we can load packages with 0 extra loading time. For this we first need to rearrange [`downloadPackage`](https://github.com/pyodide/pyodide/blob/main/src/js/load-package.ts#L159) so that it can wait until Python is ready to do the integrity checks if needed. This should be fine because we just want to get the requests started before Python is loaded, it's fine to wait to do the integrity checks.

**Comment by rth** [comment]
Indeed, I see your point. But this is more about the fact of using PyPi, micropip with a list of URLs would be equivalent. So saying "[pyodide.loadPackage](https://pyodide.org/en/latest/usage/api/js-api.html#pyodide.loadPackage) is much faster" is not very accurate.

**Comment by hoodmane** [comment]
Well sure. I really meant that it's faster with a lock file.
